splinterd/src/daemon.rs

```diff
@@ -252,26 +255,28 @@ impl SplinterDaemon {
 
         // this thread will just be dropped on shutdown
         let _ = thread::spawn(move || {
-            for connection_result in network_listener.incoming() {
-                let connection = match connection_result {
-                    Ok(connection) => connection,
-                    Err(AcceptError::ProtocolError(msg)) => {
-                        warn!("Failed to accept connection due to {}", msg);
-                        continue;
-                    }
-                    Err(AcceptError::IoError(err)) => {
-                        error!("Unable to receive new connections; exiting: {:?}", err);
-                        return Err(StartError::TransportError(format!(
-                            "Accept Error: {:?}",
-                            err
-                        )));
-                    }
-                };
-                debug!("Received connection from {}", connection.remote_endpoint());
-                match network_clone.add_connection(connection) {
-                    Ok(peer_id) => debug!("Added connection with ID {}", peer_id),
-                    Err(err) => error!("Failed to add connection to network: {}", err),
-                };
+            for mut network_listener in network_listeners {
+                for connection_result in network_listener.incoming() {
```

Comment:
Doesn't this block? so you are only going to be listening to one network listener?

Reply:
Updated so each listener has its own thread